shellfuncs/core.py

```diff
@@ -66,15 +66,25 @@ class ShellScriptFinder:
     """
     @classmethod
     def find_spec(cls, name, path=None, target=None):
-        script_path = Path('{0}.sh'.format(name.replace('.', '/')))
-        if not script_path.exists():  # no such script found.
-            return None
-
-        # TODO: check relative imports and directory walking
-        loader = ShellScriptLoader()
-        spec = importlib.util.spec_from_loader(name, loader)
-        spec.script = script_path
-        return spec
+        # import is of the form: `from path.to.script import`
+        if len(name.split('.')) > 1:
```

Comment:
I really like the idea here. The problem will be that the FS supports characters in path components which are not valid Python identifiers. 
Would you be up to convert every character in `name` which is not a valid Python identifier to a `_`.
See https://docs.python.org/3/reference/lexical_analysis.html#identifiers for valid Python identifiers :)

Reply:
Indeed I agree this was probably the most fragile bit of the PR. I definitely didn't have time to come up with the proper solution for the scope of what I needed currently, but I certainly believe I can, and should, easily at least come up with a few rudimentary sanity checks before accepting random things :D